operators/test/e2e/failure_test.go

```diff
@@ -99,18 +104,137 @@ func TestKillOneMasterNode(t *testing.T) {
 }
 
 func TestKillSingleNodeReusePV(t *testing.T) {
-	// TODO :)
-	// This test cannot work until we correctly reuse PV between pods in the operator.
-	// We should not loose data, and ClusterUUID should stay the same
-
-	// s := stack.NewStackBuilder("test-failure-kill-single-node-no-pv").
-	// 	WithESMasterDataNodes(1, stack.DefaultResources).
-	//  WithPV().
-	// 	Stack
-	// matchNode := func(p corev1.Pod) bool {
-	// 	return true // match first node we find
-	// }
-	// killNodeTest(t, s, matchNode)
+	s := stack.NewStackBuilder("test-failure-pvc").
+		WithESMasterDataNodes(1, stack.DefaultResources)
+	matchNode := func(p corev1.Pod) bool {
+		return true // match first node we find
+	}
+	killNodeTest(t, s, helpers.ESPodListOptions(s.Elasticsearch.Name), matchNode)
+}
+
+// TestKillCorrectPVReuse sets up a cluster with multiple PVs per node, kills a node, then makes sure that:
+// - PVC are reused with the correct volume (eg. do not bind the "data" PVC to a "non-data" volume)
+// - if no PVC is available, a new one is created
+func TestKillCorrectPVReuse(t *testing.T) {
+	helpers.MinVersionOrSkip(t, "v1.12.0")
+
+	k := helpers.NewK8sClientOrFatal()
+
+	lateBinding := v1.VolumeBindingWaitForFirstConsumer
```

Comment:
This bit seems tricky enough to deserve some comments:
```
// When working with multiple PVs on a single pod, there's a risk each PV get assigned to a different zone,
// not taking into consideration pod scheduling constraints. As a result, the pod becomes unschedulable.
// This is a Kubernetes issue, that can be dealt with relying on storage classes with `volumeBindingMod: WaitForFirstConsumer`.
// With this binding mode, the pod is scheduled before its PVs, which then take into account zone constraints.
// That's the only way to work with multiple PVs. Since the k8s cluster here may have a default storage class
// with `volumeBindingMode: Immediate`, we create a new one, based on the default storage class, that uses
// `waitForFirstConsumer`.
```